src/neighbors.cc

```diff
@@ -1756,27 +1756,30 @@ neighborsHtcpReply(const cache_key * key, HtcpReplyData * htcp, const Ip::Addres
     mem->ping_reply_callback(p, ntype, AnyP::PROTO_HTCP, htcp, mem->ircb_data);
 }
 
+static bool
+neighborsHtcpClearPeerNeeded(const CachePeer &p, const htcp_clr_reason reason)
+{
+    if (!p.options.htcp)
+        return false;
+    if (p.options.htcp_no_clr)
+        return false;
+    if (p.options.htcp_no_purge_clr && reason == HTCP_CLR_PURGE)
+        return false;
+    return true;
+}
+
 /*
  * Send HTCP CLR messages to all peers configured to receive them.
  */
 void
 neighborsHtcpClear(StoreEntry * e, const char *uri, HttpRequest * req, const HttpRequestMethod &method, htcp_clr_reason reason)
 {
-    CachePeer *p;
-    char buf[128];
-
-    for (p = Config.peers; p; p = p->next) {
-        if (!p->options.htcp) {
-            continue;
-        }
-        if (p->options.htcp_no_clr) {
-            continue;
-        }
-        if (p->options.htcp_no_purge_clr && reason == HTCP_CLR_PURGE) {
-            continue;
+    for (CachePeer *p = Config.peers; p; p = p->next) {
+        if (neighborsHtcpClearPeerNeeded(*p, reason)) {
+            static char buf[128];
+            debugs(15, 3, "neighborsHtcpClear: sending CLR to " << p->in_addr.toUrl(buf, 128));
```

Comment:
please remove the "neighborsHtcpClear:" part of this with the alteration.

Reply:
Fixed.